Gems/AWSClientAuth/Code/Source/AWSClientAuthSystemComponent.cpp

```diff
@@ -169,10 +169,29 @@ namespace AWSClientAuth
         AZ::Interface<IAWSClientAuthRequests>::Register(this);
         AWSClientAuthRequestBus::Handler::BusConnect();
 
-        // Objects below depend on bus above.
         m_authenticationProviderManager = AZStd::make_unique<AuthenticationProviderManager>();
-        m_awsCognitoUserManagementController = AZStd::make_unique<AWSCognitoUserManagementController>();
-        m_awsCognitoAuthorizationController = AZStd::make_unique<AWSCognitoAuthorizationController>();
+
+        // Sanity check if code should setup Cognito user and autorization controllers.
+        // Only set up if Cognito settings appear to be provided in resource mapping file.
+        AZStd::string userPoolId;
+        AWSCore::AWSResourceMappingRequestBus::BroadcastResult(
+            userPoolId, &AWSCore::AWSResourceMappingRequests::GetResourceNameId, CognitoUserPoolIdResourceMappingKey);
+
+        AZStd::string cognitoIdentityPoolId;
+         AWSCore::AWSResourceMappingRequestBus::BroadcastResult(
+            cognitoIdentityPoolId, &AWSCore::AWSResourceMappingRequests::GetResourceNameId, CognitoIdentityPoolIdResourceMappingKey);
+
+        if (userPoolId.empty() && cognitoIdentityPoolId.empty())
+        {
+            AZ_Warning("AWSClientAuthSystemComponent",  false,
```

Comment:
Nit: I'm wondering whether this warning will cause any more confusion in the log. Developers may just enable the gem without the resources deployed or implement their own controllers. Maybe we can just use AZ_TracePrintf instead here?

Reply:
I think having a warning here is fine as you have the AWS Client Auth Gem enabled, but not correctly configured, so warning to me seems like right thing todo. Most user will only ever see this if they are working with AutomatedTesting or have explicitly included the Gem, so warning appears valid (ie you have work to do here).